For #4779 : Don't double navigate with global directions in HomeActivity

ekager opened the pull request:
### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

NotWoods commented on app/src/main/java/org/mozilla/fenix/HomeActivity.kt:
```diff
@@ -221,22 +240,42 @@ open class HomeActivity : AppCompatActivity(), ShareFragment.TabsSharedCallback
-            "settings_delete_browsing_data" -> NavGraphDirections.actionGlobalDeleteBrowsingDataFragment()
+        // Make sure we aren't double navigating when using global directions by checking the destination
+        val (directions, destinationId) = when (link) {
+            "home" -> Pair(NavGraphDirections.actionGlobalHomeFragment(), R.id.homeFragment)
```
Could this be abstracted to a helper like `BrowserDirection`?

NotWoods commented on app/src/main/java/org/mozilla/fenix/ext/NavController.kt:
```diff
@@ -13,27 +13,44 @@ import androidx.navigation.Navigator
 
-fun NavController.nav(@IdRes id: Int?, directions: NavDirections) {
-    if (id == null || this.currentDestination?.id == id) {
+fun NavController.nav(
```
nit: Since the nav helper is starting to get a little complicated some KDoc comments would be nice

NotWoods commented on app/src/main/java/org/mozilla/fenix/GlobalDirections.kt:
```diff
@@ -0,0 +1,47 @@
+ * @property directions Pair which includes the directions to navigate and the fragment Id of the
+ * fragment being navigated to
+ */
+enum class GlobalDirections(val directions: Pair<NavDirections, Int>) {
```
Could there be two properties instead of a pair?

ekager commented:
@NotWoods I changed a lot here but I think it's a good start to clean up some of the navigation. I also added the directions for browser back into this PR. Let's chat tomorrow about what you need differently for custom tabs. There are still bugs with navigation (that exist in release) like I noticed because we pop to a browserfragment if it exists in openToBrowser to avoid the "display already acquired" crash, we then are breaking "open link from external app -> press back" should always take you back to the original app.

NotWoods commented on app/src/main/java/org/mozilla/fenix/GlobalDirections.kt:
```diff
@@ -0,0 +1,39 @@
+    SearchEngine(
+        NavGraphDirections.actionGlobalSearchEngineFragment(),
+        R.id.searchEngineFragment
+
```
nit: remove extra lines
